### PR TITLE
Define a trait for Intl.RelativeTimeFormat

### LINKED ISSUE #182: Implement `Intl.RelativeTimeFormat`
See here:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/RelativeTimeFormat/RelativeTimeFormat

### PR DESCRIPTION
**Pull request by filmil**

### CODE REVIEW

**Comment by filmil** [comment]
@zbraniecki PTAL?

**Review by kpozin — APPROVED** [review]

**Comment by kpozin** [comment]
Title nit: Maybe "Define a trait" instead of "Implement a trait"? When I see "Implement a trait," I think `impl FooTrait for Bar`.

File: ecma402_traits/src/lib.rs [hunk 1]
```diff
@@ -48,3 +48,9 @@ pub mod collator;
 ///
 /// The [datetimeformat] mod contains all the needed implementation bits for `Intl.DateTimeFormat`.
 pub mod datetimeformat;
+
+/// A Rust implementation of ECMA 402 RelativeTimeFormat API.
+///
+/// The [relativetimeformat] mod contains all the needed implementation bits for
+/// `Intl.RelativeTimeFormat`.
+pub mod relativetimeformat;
```
Comment (zbraniecki) [inline ecma402_traits/src/lib.rs:56]: Nit: I'd use datetime.rs, relative time.rs etc. And only name types DateTimeFormat and RelativeTimeFormat

File: ecma402_traits/src/relativetimeformat.rs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+/// options in the JS version.
+///
+/// See [Options] for the contents of the options.  See the [RelativeTimeFormat::try_new]
+/// for the use of the options.
+pub mod options {
+    /// Whether to use numeric formatting in the display, like "1 day ago".
+    #[derive(Debug, Clone, PartialEq)]
+    pub enum Numeric {
+        /// Always use numeric formatting.
+        ALWAYS,
```
Comment (zbraniecki) [inline ecma402_traits/src/relativetimeformat.rs:31]: If I'm not mistakes cargo clippy recommends Always, not ALWAYS

**Review by zbraniecki — APPROVED** [review]

File: ecma402_traits/src/relativetimeformat.rs (hunk 2 above)
Comment (filmil) [reply ecma402_traits/src/relativetimeformat.rs]: Done.

File: ecma402_traits/src/lib.rs (hunk 1 above)
Comment (filmil) [reply ecma402_traits/src/lib.rs]: Done.

**Comment by filmil** [comment]
> Title nit: Maybe "Define a trait" instead of "Implement a trait"? When I see "Implement a trait," I think `impl FooTrait for Bar`.

Done

**Pull request merged**